### ISSUE TITLE
Unused variables inside if-let guards don't get detected

### ISSUE THREAD

**Issue by fmease** [rust-lang/rust#119383]
I expected the following code to fail to compile due to the unused variable `unused` but it actually compiled successfully:

```rs
#![feature(if_let_guard)]
#![deny(unused_variables)]

fn main() {
    match () {
        () if let Some(unused) = Some(()) => {}
        _ => {}
    }
}
```
